ssc/cmod_wind_landbosse.cpp

```diff
@@ -0,0 +1,350 @@
+	STARTUPINFO si;
+	SECURITY_ATTRIBUTES sa;
+	PROCESS_INFORMATION pi;
+    HANDLE g_hChildStd_IN_Rd = NULL;
+    HANDLE g_hChildStd_OUT_Wr = NULL;
+    HANDLE g_hChildStd_OUT_Rd = NULL;
+    HANDLE g_hChildStd_IN_Wr = NULL;
+    HANDLE stderr_rd = NULL;
+    HANDLE stderr_wr = NULL;  //pipe handles
+	char buf[BUFSIZE];           //i/o buffer
+	memset(buf, 0, sizeof(buf));
+
+	std::string pythonpath = std::string(get_python_path()) + "\\" + python_exec_path;
+	CA2T programpath( pythonpath.c_str());
+	std::string pythonarg = " -c \"" + python_run_cmd + "\"";
+	size_t pos = pythonarg.find("<input>");
+	pythonarg.replace(pos, 7, input_dict_as_text);
+	CA2T programargs(pythonarg.c_str());
+
+	sa.nLength = sizeof(SECURITY_ATTRIBUTES);
+	sa.bInheritHandle = TRUE;
+	sa.lpSecurityDescriptor = NULL;
+
+	std::string output;
+    if (!CreatePipe(&g_hChildStd_IN_Rd, &g_hChildStd_IN_Wr, &sa, 0)) {
+        goto error;
+    }
+    if (!CreatePipe(&g_hChildStd_OUT_Rd, &g_hChildStd_OUT_Wr, &sa, 0)) {
+        goto error;
+    }
+    if (!CreatePipe(&stderr_rd, &stderr_wr, &sa, 0)) {
+        goto error;
+    }
+            
+    //set startupinfo for the spawned process
+    /*The dwFlags member tells CreateProcess how to make the process.
+    STARTF_USESTDHANDLES: validates the hStd* members.
+    STARTF_USESHOWWINDOW: validates the wShowWindow member*/
+    GetStartupInfo(&si);
+
+    si.dwFlags = STARTF_USESTDHANDLES | STARTF_USESHOWWINDOW;
+    si.wShowWindow = SW_HIDE;
+    //set the new handles for the child process
+    si.hStdOutput = g_hChildStd_OUT_Wr;
+    si.hStdError = stderr_wr;
+    si.hStdInput = g_hChildStd_IN_Rd;
+
+    //spawn the child process
+    if (CreateProcess(programpath, programargs, NULL, NULL, TRUE, CREATE_NO_WINDOW,
+        NULL, NULL, &si, &pi))
+    {
+        unsigned long bread;   //bytes read
+        unsigned long bread_last = 0;
+        unsigned long avail;   //bytes available
+
+        for (;;)
+        {
+            PeekNamedPipe(g_hChildStd_OUT_Rd, buf, BUFSIZE - 1, &bread, &avail, NULL);
+            //check to see if there is any data to read from stdout
+            if (bread != 0)
+            {
+                if (ReadFile(g_hChildStd_OUT_Rd, buf, BUFSIZE - 1, &bread, NULL))
+                {
+                    output += std::string(buf);
+                    bread_last = bread;
+                }
+            }
+            else if (bread_last > 0)
+            {
+                break;
+            }
+        }
+
+        CloseHandle(pi.hThread);
+        CloseHandle(pi.hProcess);
+    }
+   error:
+    std::vector<HANDLE> handles = {g_hChildStd_IN_Rd, g_hChildStd_IN_Wr, g_hChildStd_OUT_Rd, g_hChildStd_OUT_Wr, stderr_rd, stderr_wr};
+    for (HANDLE handle : handles) {
+        if (handle && handle != INVALID_HANDLE_VALUE) {
+            CloseHandle(handle);
+        }
+    }
+	return buf;
+}
+#endif
+
+void cm_wind_landbosse::cleanOutputString(std::string& output_json) {
```

Comment:
Can we push this logic to LandBOSSE?  Or make a wrapper function to JSON-ify the dictionary?  There could be other problems in the future if they add booleans (Python uses True/False instead of true/false).

Reply:
Yes, I think that's a good idea to have LandBOSSE use the json module and only return a valid json string.

@parangat94 Would it be possible to have this in the next version? Instead of returning a dictionary, could you convert it via the json module to a valid json string, to be returned from `run_landbosse`?

Reply:
Another option is for the run function to accept a `json` parameter that tells it to format the output as JSON.  Then it could support dictionaries (for native Python callers) or JSON.

Reply:
I'll add this to the "run_cmd" in landbosse.json

Reply:
@parangat94 But the issue is that LandBOSSE prints a failure message. This is what gets read from std out:

```
"Fail SAM_Run FoundationCost\n{\"errors\": [\"Error in NegativeInputError: User entered a negative value for depth. This is an invalid entry\", \"Error in FoundationCost: cannot convert float NaN to integer\"]}\n"
```

We want it to just say
```
"{\"errors\": [\"Error in NegativeInputError: User entered a negative value for depth. This is an invalid entry\", \"Error in FoundationCost: cannot convert float NaN to integer\"]}\n"
```

Reply:
Adding this to Jira and resolving for now.

Reply:
@parangat94 Can you make LandBOSSE write all error messages to stderr instead of stdout?  Basically, you would add `, file=sys.stderr` to all your calls to `print` when there is an error.